spec/features/rooms/index_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'index view' do
+  let!(:holliday) { Hotel.create!(name: 'Holliday Inn', location: 'Buxton') }
+  let!(:harbor) { Hotel.create!(name: 'Portland Harbor', location: 'Portland') }
+  let!(:room_1) { harbor.rooms.create!(rate: 180, suite: "Presidential")}
+  let!(:room_2) { harbor.rooms.create!(rate: 185, suite: "Vice-Presidential")}
+  let!(:room_3) { holliday.rooms.create!(rate: 160, suite: "Honeymoon")}
+  let!(:room_4) { holliday.rooms.create!(rate: 10, suite: "Broom Closet")}
+  let!(:steve) { room_1.guests.create!(name: 'Steve', nights: 3) }
+  let!(:dave) { room_2.guests.create!(name: 'Dave', nights: 5) }
+  let!(:jim) { room_3.guests.create!(name: 'Jim', nights: 1)}
+
+  it 'lists all rooms and associated attributes' do
+    visit '/rooms'
+
+    GuestRoom.create!(guest: steve, room: room_1)
+    GuestRoom.create!(guest: dave, room: room_1)
+
+    GuestRoom.create!(guest: steve, room: room_2)
+
+    GuestRoom.create!(guest: dave, room: room_3)
+
+    GuestRoom.create!(guest: dave, room: room_4)
+
+    expect(page).to have_content(room_1.suite)
+    expect(page).to have_content(room_1.rate)
+    expect(page).to have_content(harbor.name)
+    expect(page).to have_content(room_1.guests.count)
```

Comment:
This test is failing. Whenever the test reaches line `29` it fails because it does not find the number `3` on the page, `3` being the number of guests who have rented room 1.

The issue here is ordering, because we `visit` the page before the `GuestRoom` associations get created, there is only 1 guest who has stayed in the room at the time of the request. The solution is to move the `visit` to after the `GuestRoom` associations.

```ruby
GuestRoom.create!(guest: steve, room: room_1)
GuestRoom.create!(guest: dave, room: room_1)

GuestRoom.create!(guest: steve, room: room_2)

GuestRoom.create!(guest: dave, room: room_3)

GuestRoom.create!(guest: dave, room: room_4)

visit '/rooms'
```

By this time, when the request is made, we would have those associations in the database.